Add RBAC files for different deployment strategies

bhavin192 opened the pull request:
Related to https://github.com/yugabyte/yugabyte-db/issues/6348

Scenarios tested:
- Created a new providers with kubeconfig created using
  platform-global.yaml and platform-namespaced.yaml.
- Tried creating a new universe, edited it (changed number of nodes),
  took backup, deleted the universe.
- All the above operations worked just fine.

iSignal commented on stable/yugabyte/templates/service.yaml:
```diff
@@ -1,5 +1,27 @@
+data:
+{{- $rootCAClient := buildCustomCert $root.Values.tls.rootCA.cert $root.Values.tls.rootCA.key -}}
+{{- $client := genSignedCert "yugabyte" ( default nil ) ( default nil ) 3650 $rootCAClient }}
+  root.crt: {{ $rootCAClient.Cert | b64enc }}
```
root.crt is named ca.crt in the docs. Does it work with either name?

iSignal commented:
looks good to me! just a minor comment. if you tested this already, feel free to ignore and check it in.

bhavin192 committed:
Add RBAC files for different deployment strategies

Scenarios tested:
- Created a new providers with kubeconfig created using
  platform-global.yaml and platform-namespaced.yaml.
- Tried creating a new universe, edited it (changed number of nodes),
  took backup, deleted the universe.
- All the above operations worked just fine.

Signed-off-by: Bhavin Gandhi <bhavin7392@gmail.com>

iSignal commented on rbac/README.md:
```diff
@@ -0,0 +1,86 @@
+## RBAC resources for Yugabyte Platform
+
+This directory contains different RBAC manifests containing Roles,
+RoleBindings etc required for the Yugabyte Platform.
```
required to create yugabyte universes using Yugabyte Platform

iSignal commented on rbac/README.md:
```diff
@@ -0,0 +1,86 @@
+
+### 0. yugabyte-platform-universe-management-sa.yaml (required)
+This is the ServiceAccount whose secret can be used to generate a
+kubeconfig. This should be created in a namespace where your are going
```
This should ideally be created in a namespace

iSignal commented on rbac/README.md:
```diff
@@ -0,0 +1,86 @@
+following steps depending to your requirements.
+
+### a. platform-global.yaml
+Grants access to manage YugabyteDB universes across all the namespaces
```
Grants access to only the specific cluster roles to create and manage YugabyteDB

iSignal commented on rbac/README.md:
````diff
@@ -0,0 +1,86 @@
+```
+
+### b. platform-global-admin.yaml
+Grants cluster level admin access.
````
Grants broad cluster level admin access.

iSignal commented on rbac/README.md:
````diff
@@ -0,0 +1,86 @@
+```
+
+### c. platform-namespaced.yaml
+Grants access to manage YugabyteDB universes in a particular namespace
````
Grants access to only the specific roles required to create and manage ...

iSignal reviewed (approved):
Thanks for identifying the specific roles needed, both at ns and cluster level. 

Once we get these integrated with itest, we will be able to keep them in sync with the code.
